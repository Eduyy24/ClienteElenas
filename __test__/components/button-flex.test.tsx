import React from 'react';
import { create } from 'react-test-renderer';
import Button from '../../src/components/button-flex';
import { fireEvent, render } from '@testing-library/react-native'


describe('Button Snapshot', () => {

  test('Comprobar la UI del componente Button', () => {
    const footer = create(<Button onPress={() => {}} title="CREAR" />);
    expect(footer.toJSON()).toMatchSnapshot();
  });

  test('examples of some things', async () => {
    const onPress = jest.fn()
    const { getByText } = render(<Button onPress={onPress} title="CREAR" />)
  
    const button = getByText('CREAR')
    fireEvent.press(button)

    expect(onPress).toHaveBeenCalledTimes(1);
  })
});