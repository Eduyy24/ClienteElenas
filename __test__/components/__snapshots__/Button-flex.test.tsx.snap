// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Snapshot Comprobar la UI del componente Button 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#A2A7A8",
      "flex": 1,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    CREAR
  </Text>
</View>
`;
